sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.cs

```diff
@@ -892,6 +892,15 @@ string Local()
                 return buffer;
             }
         }
+    }
 
+    // https://github.com/SonarSource/sonar-dotnet/issues/3304
+    public class Repro_3304
+    {
+        public void WithRefKeyword(int[] values)
+        {
+            ref int value = ref values[0]; // Noncompliant FP because there exists write that is propagated
```

Comment:
```suggestion
            ref int value = ref values[0]; // Noncompliant FP because `value` keeps the reference to `values[0]`, and below `default` is actually assigned to `values[0]`
```